Refactoring

slivering opened the pull request.

slivering committed:
Refactor LFighter

slivering committed:
Refactor FABA

slivering committed:
Use loguru instead of print

slivering committed:
Refactor StaticLabelFlipping

matol-16 commented on byzfl/aggregators/aggregators.py:
```diff
@@ -656,88 +656,107 @@ def __call__(self, vectors):
+
+        return avg
 
+    def __call__(self, local_models, global_model):
```
ici, le format avec 3 arguments crée une erreur quand l'aggrégateur est appelé dans train.py via une méthode de la classe robust_aggregator. On peut s'en passer en appelant directement l'aggrégateur dans update_model_with_models

slivering replied:
Il me semble que le cas spécial de LFighter est géré par ces deux lignes :

https://github.com/EA-Robust-AI-X2023/byzfl/blob/a83a2a59459c4ebbb20e817b56839833b444f334/byzfl/benchmark/train.py#L361

https://github.com/EA-Robust-AI-X2023/byzfl/blob/a83a2a59459c4ebbb20e817b56839833b444f334/byzfl/benchmark/train.py#L423

Est-ce que tu peux préciser l'erreur exacte et comment la reproduire ?

matol-16 replied:
L'erreur vient de server.py l94:        ` aggregate_weights = self.aggregate(models, global_model)`

En effet, ici, le self.aggregate doit recevoir un seul argument noramelement

matol-16 commented on byzfl/aggregators/aggregators.py:
```diff
@@ -656,88 +656,107 @@ def __call__(self, vectors):
+        global_model = list(global_model.parameters())
+        
+        dw = np.array([
+            global_model[i][-2].numpy(force=True) - local_models[i][-2].numpy(force=True)
```
Ici j'ai l'erreur (sans doute car en régression softmax, on a pas assez de layers pour faire [i][-2]

File "/users/eleves-b/2023/mathias.ollu/mean_robustness/.venv/lib/python3.12/site-packages/byzfl/aggregators/aggregators.py", line 716, in __call__
    global_model[i][-2].numpy(force=True) - local_models[i][-2].numpy(force=True)
    ~~~~~~~~~~~~^^^
IndexError: list index out of range
"""

slivering replied:
Ça me semble pas normal : avec `softmax_mnist` il y a exactement 2 paramètres (W et b) :
```python
from byzfl.fed_framework.models import softmax_mnist
model = softmax_mnist()
[param.shape for param in model.parameters()]
```

```
[torch.Size([10, 784]), torch.Size([10])]
```

slivering committed:
Update dependencies

slivering committed:
Fix update_model_with_models for LFighter

slivering committed:
Use dict gradients to factor in momentum

slivering committed:
FIx typos "poisonned" -> "poisoned"

matol-16 committed:
fix faba device bug

matol-16 committed:
fix: import error

slivering committed:
fix: reconciliate LFighter and momentum

Two steps to be consistent with the paper:

1. apply LFighter on the gradients w/o momentum to identify poisoned gradients
2. aggregate the honest gradients w/ momentum

matol-16 committed:
fix : lfighter last issues after test

matol-16 committed:
Merge branch 'variance_and_plot' into refactoring
